examples/simple-regression/src/model.rs

```diff
@@ -0,0 +1,80 @@
+use crate::dataset::DiabetesBatch;
+use burn::{
+    module::Module,
+    nn::{loss::MSELoss, Linear, LinearConfig},
+    tensor::{
+        backend::{AutodiffBackend, Backend},
+        Tensor,
+    },
+    train::{RegressionOutput, TrainOutput, TrainStep, ValidStep},
+};
+use burn::nn::loss::Reduction::Mean;
+use burn::nn::ReLU;
+
+#[derive(Module, Debug)]
+pub struct LinearModel<B: Backend> {
+    input_layer: Linear<B>,
+    output_layer: Linear<B>,
+    activation: ReLU
+}
+
+impl<B: Backend> Default for LinearModel<B> {
+    fn default() -> Self {
+        let device = B::Device::default();
+        Self::new(10, &device)
+    }
+}
+
+
+impl<B: Backend> LinearModel<B> {
+    pub fn new(feature_len: usize, device: &B::Device) -> Self {
+        let input_layer = LinearConfig::new(feature_len, 64)
+            .with_bias(true).init(device);
+        let output_layer = LinearConfig::new(64, 1)
+            .with_bias(true).init(device);
+
+        Self {
+            input_layer,
+            output_layer,
+            activation: ReLU::new()
+        }
+    }
```

Comment:
Instead of a new function, I would create a config:

```rust
#[derive(Config)]
pub struct RegressionModelConfig {
    pub num_features: usize;
    #[config(default = 64)]
    pub hidden_size: usize;
}
```
Then I would implement init

```rust
impl RegressionModelConfig {
    pub fn init<B: Backend>(&self, device: &B::Device) -> RegressionModel<B> {
            let input_layer = LinearConfig::new(self.num_features, self.hidden_size)
            .with_bias(true).init(device);
        let output_layer = LinearConfig::new(self.hidden_size, 1)
            .with_bias(true).init(device);

        Self {
            input_layer,
            output_layer,
            activation: ReLU::new()
        }
    }
}
```

Something like that.